store/store.go

```diff
@@ -1119,6 +1128,31 @@ func findRev(needle snap.Revision, haystack []snap.Revision) bool {
 // The file is saved in temporary storage, and should be removed
 // after use to prevent the disk from running out of space.
 func (s *Store) Download(name string, downloadInfo *snap.DownloadInfo, pbar progress.Meter, user *auth.UserState) (path string, err error) {
+
+	if useDeltas() {
+		logger.Debugf("Available deltas returned by store: %v", downloadInfo.Deltas)
+	}
+	if useDeltas() && len(downloadInfo.Deltas) == 1 {
+		workingDir, err := ioutil.TempDir("", name+"-deltas")
+		deltaInfo := &downloadInfo.Deltas[0]
+		if err == nil {
```

Comment:
This `err == nil {` cascading is a bit of an uncommon pattern in go. Usually its `if err != nil { return err }` or something. Maybe you can move this into a helper  like downloadAndApplyDelta() and just return with err if any of the steps fails?